### PR TITLE
Merkle endpoints for state proof 

### PR DESCRIPTION
**Pull request by fboucquez**
Changes:

1. /merkle endpoints
2. get id by composite hash
3. hash and secret lock new searches
4. Renamed Status to LockStatus enum

POC, MerkleInfo still to be defined. 

Related: https://github.com/nemtech/catapult-rest/pull/519

### CODE REVIEW

File: spec/schemas/MerkleInfo.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+type: object
```
Comment (segfaultxavi) [inline spec/schemas/MerkleInfo.yml:1]: No `description` here? What is this string?

**Review by segfaultxavi — CHANGES_REQUESTED** [review]

File: spec/schemas/MerkleInfo.yml (hunk 1 above)
Comment (fboucquez) [reply spec/schemas/MerkleInfo.yml:1]: This is the missing part. I still don't know how to parse the server API merkle path. Once I have it this object will be modified and documented.

Comment (segfaultxavi) [reply spec/schemas/MerkleInfo.yml:1]: 👍

Comment (fboucquez) [reply spec/schemas/MerkleInfo.yml]: Hi @segfaultxavi . I've updated the information with the best information I have right now. In the future, we will upgrade this object adding a better structure for state proof.

Comment (fboucquez) [reply spec/schemas/MerkleInfo.yml]: Hi @segfaultxavi  the raw string is the best I can return right now. In the future, the object will also have a list describing the nodes in the tree. Something like: 

```
[
  { type: branch, path: ..., linkMask: xxx, links: [ ... , ... ] },
  { type: leaf, path: ..., hash }
]
```

**Review by rg911 — APPROVED** [review]

**Pull request merged**